Source/Synthesis/Generation/MetadataGenerator.cs

```diff
@@ -59,7 +60,10 @@ public TemplateGenerationMetadata GenerateMetadata()
 
 						var fieldInfo = new FieldPropertyInfo(field);
 						fieldInfo.FieldPropertyName = propertyName;
-						fieldInfo.SearchFieldName = _indexFieldNameTranslator.GetIndexFieldName(field.Name);
+						if (_indexFieldNameTranslator.GetType().Name.ToLower().Contains("solr"))
+							SolrFieldMappingHack(field, fieldInfo);
+						else
+							fieldInfo.SearchFieldName = _indexFieldNameTranslator.GetIndexFieldName(field.Name);
```

Comment:
I'd prefer to see this logic moved out of the generator, as SOLR should not be a concern here. It looks like we might be able to extract the hacks into some kind of wrapper class that inherits from the Solr field name translator?

e.g. SynthesisSolrFieldNameTranslator : SolrFieldNameTranslator, then get GetIndexFieldName() to operate correctly.